### PR TITLE
Denormalize keypoints coordinates

### PR DESCRIPTION
**Pull request by gfloros**

### CODE REVIEW

File: hloc/colmap_from_opensfm.py [hunk 1]
```diff
@@ -41,6 +41,16 @@ def angle_axis_to_quaternion(angle_axis: np.ndarray) -> list[float]:
     return np.array([qw, float(qx), float(qy), float(qz)])
 
 
+def denormalize_image_coordinates(
+    norm_coords: np.ndarray, width: int, height: int
+) -> np.ndarray:
+    size = max(width, height)
+    p = np.empty((len(norm_coords), 2))
+    p[:, 0] = norm_coords[:, 0] * size - 0.5 + width / 2.0
+    p[:, 1] = norm_coords[:, 1] * size - 0.5 + height / 2.0
```
Comment (blaketroutmandd) [inline hloc/colmap_from_opensfm.py:50]: Are the order of operations correct here? Wouldn't `0.5` need to be subtracted from `norm_coords` before scaling up its size and adding half of the width/height back to it?

Comment (gfloros) [reply hloc/colmap_from_opensfm.py:50]: The order is correct. We subtract half a pixel after we have denormalized (multiplying by size).
See [here](https://github.com/dronedeploy/OpenSFM/blob/0d596597fbeaecd62dc6ae228e7ac06b75a87c8b/opensfm/features.py#L315-L332) the normalization/denormalization functions of OpenSfM. Actually it's a copy of this function.

Comment (blaketroutmandd) [reply hloc/colmap_from_opensfm.py:50]: Ahh I see, the `0.5` _is_ supposed to represent half of a pixel.

**Review by blaketroutmandd — APPROVED** [review]
LGTM

**Review by blaketroutmandd — APPROVED** [review]
LGTM

**Pull request merged**